tracer/test/Datadog.Trace.Tests/Configuration/ExporterSettingsTests.cs

```diff
@@ -0,0 +1,158 @@
+            Assert.Equal(expected: new Uri($"http://127.0.0.1:8126"), actual: config.AgentUri);
+            Assert.Equal(expected: 8125, actual: config.DogStatsdPort);
+            Assert.False(config.PartialFlushEnabled);
+            Assert.Equal(expected: 500, actual: config.PartialFlushMinSpans);
+        }
+
+        [Fact]
+        public void PartialFlushVariables_Populated()
+        {
+            var config = Setup(FileExistsMock(), "DD_TRACE_PARTIAL_FLUSH_ENABLED:true", "DD_TRACE_PARTIAL_FLUSH_MIN_SPANS:999");
+            Assert.True(config.PartialFlushEnabled);
+            Assert.Equal(expected: 999, actual: config.PartialFlushMinSpans);
+        }
+
+        [Fact]
+        public void Traces_SocketFilesExist_NoExplicitConfig_UsesTraceSocket()
+        {
+            var config = Setup(DefaultSocketFilesExist());
+            Assert.Equal(expected: TracesTransportType.UnixDomainSocket, actual: config.TracesTransport);
+            Assert.Equal(expected: ExporterSettings.DefaultTracesUnixDomainSocket, actual: config.TracesUnixDomainSocketPath);
+        }
+
+        [Fact]
+        public void Traces_SocketFilesExist_ExplicitAgentHost_UsesDefaultTcp()
+        {
+            var agentHost = "someotherhost";
+            var expectedUri = new Uri($"http://{agentHost}:8126");
+            var config = Setup(DefaultSocketFilesExist(), "DD_AGENT_HOST:someotherhost");
+            Assert.Equal(expected: TracesTransportType.Default, actual: config.TracesTransport);
+            Assert.Equal(expected: expectedUri, actual: config.AgentUri);
+        }
+
+        [Fact]
+        public void Traces_SocketFilesExist_ExplicitTraceAgentPort_UsesDefaultTcp()
+        {
+            var expectedUri = new Uri($"http://127.0.0.1:8111");
+            var config = Setup(DefaultSocketFilesExist(), "DD_TRACE_AGENT_PORT:8111");
+            Assert.Equal(expected: TracesTransportType.Default, actual: config.TracesTransport);
+            Assert.Equal(expected: expectedUri, actual: config.AgentUri);
+        }
+
+        [Fact]
+        public void Traces_SocketFilesExist_ExplicitWindowsPipeConfig_UsesWindowsNamedPipe()
+        {
+            var config = Setup(DefaultSocketFilesExist(), "DD_TRACE_PIPE_NAME:somepipe");
+            Assert.Equal(expected: TracesTransportType.WindowsNamedPipe, actual: config.TracesTransport);
+            Assert.Equal(expected: "somepipe", actual: config.TracesPipeName);
+        }
+
+        [Fact]
+        public void Traces_SocketFilesExist_ExplicitUdsConfig_UsesExplicitConfig()
+        {
+            var config = Setup(DefaultSocketFilesExist(), "DD_APM_RECEIVER_SOCKET:somesocket");
+            Assert.Equal(expected: TracesTransportType.UnixDomainSocket, actual: config.TracesTransport);
+            Assert.Equal(expected: "somesocket", actual: config.TracesUnixDomainSocketPath);
+        }
+
+        [Fact]
+        public void Traces_SocketFilesExist_ExplicitConfigForAll_UsesDefaultTcp()
```

Comment:
This test asserts that a host will set TCP over all other choices. Can you also add a test to assert that Windows Named Pipes takes precedence over UDS?

Reply:
This same comment applies for metrics as well, so it's documented for both cases what the order of precedence is

Reply:
I could, but it didn't seem worth enforcing considering they are explicitly set in each situation. In the future if someone decides to change this code to prioritize UDS instead of Windows Named Pipes, it didn't seem like it changed anything valuable to me.

Reply:
Gotcha. For my own sanity, I wanted to document somewhere the order in which the settings would be applied instead of having to parse the code and I figured that a unit test would be an easy way to convey that. Could you add some sort of documentation (maybe in ExporterSettings) that outlines the current order?

Reply:
Done in `ExporterSettingsTests`